ytdlBinary is not defined

Hello, I have that error when I start my server. Don't Understand ? Any ideas ?

```
ReferenceError: ytdlBinary is not defined
    at ytdlBinary (/usr/src/app/node_modules/youtube-dl/lib/youtube-dl.js:26:22)
    at Object.<anonymous> (/usr/src/app/node_modules/youtube-dl/lib/youtube-dl.js:30:3)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/usr/src/app/routes/rest/assets.js:4:19)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at global.app_require (/usr/src/app/index.js:3:32)
    at Object.<anonymous> (/usr/src/app/routes/rest/index.js:10:23)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at global.app_require (/usr/src/app/index.js:3:32)
    at Object.<anonymous> (/usr/src/app/routes/index.js:15:21)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! api.v2@1.0.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the api.v2@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```



I rolled back to 1.12.1 and it work fine.

--- a/README.md
+++ b/README.md
@@ -11,7 +11,7 @@ If you're only interested in downloading only from youtube, you should consider 
 With [npm](https://www.npmjs.com/) do:
 
 ``` sh
-npm install @microlink/youtube-dl
+npm install youtube-dl
 ```
 
 ## Usage

--- a/lib/get-binary.js
+++ b/lib/get-binary.js
@@ -1,17 +1,17 @@
 'use strict'
 
+const { readFileSync, existsSync } = require('fs')
 const path = require('path')
-const fs = require('fs')
 
 const binPath = path.join(__dirname, '..', 'bin')
 const detailsPath = path.join(binPath, 'details')
 
 module.exports = () => {
-  if (!fs.existsSync(detailsPath)) {
+  if (!existsSync(detailsPath)) {
     throw new Error('ERROR: unable to locate `youtube-dl` at ' + binPath)
   }
 
-  const details = JSON.parse(fs.readFileSync(detailsPath))
+  const details = JSON.parse(readFileSync(detailsPath))
 
   return details.path
     ? details.path

--- a/lib/get-binary.js
+++ b/lib/get-binary.js
@@ -1,0 +1,19 @@
+'use strict'
+
+const path = require('path')
+const fs = require('fs')
+
+const binPath = path.join(__dirname, '..', 'bin')
+const detailsPath = path.join(binPath, 'details')
+
+module.exports = () => {
+  if (!fs.existsSync(detailsPath)) {
+    throw new Error('ERROR: unable to locate `youtube-dl` at ' + binPath)
+  }
+
+  const details = JSON.parse(fs.readFileSync(detailsPath))
+
+  return details.path
+    ? details.path
+    : path.resolve(__dirname, '..', 'bin', details.exec)
+}

--- a/lib/youtube-dl.js
+++ b/lib/youtube-dl.js
@@ -4,10 +4,8 @@ const execFile = require('child_process').execFile
 const streamify = require('streamify')
 const request = require('request')
 const hms = require('hh-mm-ss')
-const path = require('path')
 const http = require('http')
 const url = require('url')
-const fs = require('fs')
 
 const {
   isYouTubeRegex,
@@ -17,28 +15,11 @@ const {
   isString
 } = require('./util')
 
-const detailsPath = path.join(__dirname, '..', 'bin/details')
+let ytdlBinary = require('./get-binary')()
 
 const TEN_MEGABYTES = 1000 * 1000 * 10
 
 const execFileOpts = { maxBuffer: TEN_MEGABYTES }
-
-let ytdlBinary = (() => {
-  if (fs.existsSync(detailsPath)) {
-    const details = JSON.parse(fs.readFileSync(detailsPath))
-    return details.path
-      ? details.path
-      : path.resolve(__dirname, '..', 'bin', details.exec)
-  }
-
-  if (!fs.existsSync(ytdlBinary)) {
-    console.error(
-      'ERROR: unable to locate youtube-dl details in ' +
-        path.dirname(ytdlBinary)
-    )
-    process.exit(1)
-  }
-})()
 
 function youtubeDl (args, options, cb) {
   execFile(ytdlBinary, args, { ...execFileOpts, ...options }, function done (

--- a/lib/youtube-dl.js
+++ b/lib/youtube-dl.js
@@ -1,6 +1,6 @@
 'use strict'
 
-const execFile = require('child_process').execFile
+const { execFile } = require('child_process')
 const streamify = require('streamify')
 const request = require('request')
 const hms = require('hh-mm-ss')

--- a/package.json
+++ b/package.json
@@ -1,5 +1,5 @@
 {
-  "name": "@microlink/youtube-dl",
+  "name": "youtube-dl",
   "description": "youtube-dl driver for node",
   "homepage": "https://github.com/przemyslawpluta/node-youtube-dl#readme",
   "version": "2.0.0",

--- a/test/getInfo.js
+++ b/test/getInfo.js
@@ -9,7 +9,7 @@ vows
       topic: function () {
         'use strict'
         var video = 'http://www.youtube.com/watch?v=90AiXO1pAiA'
-        ytdl.getInfo(video, ['-f', '18/43/36'], this.callback)
+        ytdl.getInfo(video, ['--format=mp4'], this.callback)
       },
 
       'info returned': function (err, info) {
@@ -98,7 +98,11 @@ vows
         var youtube = 'http://www.youtube.com/watch?v=90AiXO1pAiA'
         var twitch =
           'https://clips.twitch.tv/RelentlessOptimisticPterodactylRitzMitz'
-        ytdl.getInfo([youtube, twitch], ['--no-warnings'], this.callback)
+        ytdl.getInfo(
+          [youtube, twitch],
+          ['--no-warnings', '--format=mp4'],
+          this.callback
+        )
       },
 
       'info returned': function (err, info) {
@@ -108,7 +112,7 @@ vows
         assert.strictEqual(info.length, 2)
 
         assert.strictEqual(info[0].id, '90AiXO1pAiA')
-        assert.strictEqual(info[0].format_id, '43')
+        assert.strictEqual(info[0].format_id, '18')
         assert.strictEqual(info[0].title, 'lol')
         assert.isString(info[0].url)
         assert.isString(info[0].thumbnail)
@@ -118,10 +122,10 @@ vows
             'Filmed in 2003 before Youtube was invented. ' +
             'This is also the original I find it hilarious that there are copycat videos!'
         )
-        assert.strictEqual(info[0]._filename, 'lol-90AiXO1pAiA.webm')
-        assert.strictEqual(info[0].width, 640)
+        assert.strictEqual(info[0]._filename, 'lol-90AiXO1pAiA.mp4')
+        assert.strictEqual(info[0].width, 480)
         assert.strictEqual(info[0].height, 360)
-        assert.strictEqual(info[0].format, '43 - 640x360 (medium)')
+        assert.strictEqual(info[0].format, '18 - 480x360 (medium)')
         assert.strictEqual(info[0]._duration_raw, 11)
         assert.strictEqual(info[0]._duration_hms, '00:00:11')
         assert.strictEqual(info[0].duration, '11')
@@ -155,7 +159,7 @@ vows
       topic: function () {
         'use strict'
         var youtube = 'http://www.youtube.com/watch?v=90AiXO1pAiA'
-        ytdl.getInfo(youtube, ['-g'], this.callback)
+        ytdl.getInfo(youtube, ['-g', '--format=mp4'], this.callback)
       },
 
       'info returned': function (err, info) {
@@ -164,7 +168,7 @@ vows
         assert.isNull(err)
         assert.isObject(info)
         assert.strictEqual(info.id, '90AiXO1pAiA')
-        assert.strictEqual(info.format_id, '43')
+        assert.strictEqual(info.format_id, '18')
         assert.strictEqual(info.title, 'lol')
         assert.isString(info.url)
         assert.isString(info.thumbnail)
@@ -174,12 +178,12 @@ vows
             'Filmed in 2003 before Youtube was invented. ' +
             'This is also the original I find it hilarious that there are copycat videos!'
         )
-        assert.strictEqual(info._filename, 'lol-90AiXO1pAiA.webm')
-        assert.strictEqual(info.format, '43 - 640x360 (medium)')
+        assert.strictEqual(info._filename, 'lol-90AiXO1pAiA.mp4')
+        assert.strictEqual(info.format, '18 - 480x360 (medium)')
         assert.strictEqual(info._duration_raw, 11)
         assert.strictEqual(info._duration_hms, '00:00:11')
         assert.strictEqual(info.duration, '11')
-        assert.strictEqual(info.width, 640)
+        assert.strictEqual(info.width, 480)
         assert.strictEqual(info.height, 360)
         assert.isArray(info.formats)
       }

--- a/test/playlist.js
+++ b/test/playlist.js
@@ -70,7 +70,7 @@ vows
         assert.strictEqual(data[1].progress, 1)
         assert.strictEqual(
           data[1].data._filename,
-          'LA REGAÑADA DEL MILENIO.wmv-SITuxqDUjPI.webm'
+          'LA REGAÑADA DEL MILENIO.wmv-SITuxqDUjPI.mp4'
         )
 
         function fileExists (data) {

